src/aleinterface.jl

```diff
@@ -436,7 +436,7 @@ saveScreenPNG(ale::ALEPtr, filename::String) = ccall((:saveScreenPNG, libale_c),
 
 function encodeState(state::ALEStatePtr)
```

Comment:
We can also add an in-place method `encodeState!` to avoid allocating every time.

Reply:
Are you suggesting including `encodeState` into the `cloneSystemState` and `cloneState`methods?

Reply:
I've made a patch 😄